agent/engine/docker_task_engine_test.go

```diff
@@ -1513,3 +1513,91 @@ func TestCreateContainerOnAgentRestart(t *testing.T) {
 		t.Error("Unexpected error", metadata.Error)
 	}
 }
+
+// TestTaskUseExecutionRolePullECRImage tests the agent will use the execution role
+// credentials to pull from ecr repository
+func TestTaskUseExecutionRolePullECRImage(t *testing.T) {
+	ctrl, client, _, taskEngine, credentialsManager, imageManager := mocks(t, &defaultConfig)
+	defer ctrl.Finish()
+
+	credentialsID := "execution role"
+	accessKeyID := "akid"
+	secretAccessKey := "sakid"
+	sessionToken := "token"
+	executionRoleCredentials := credentials.IAMRoleCredentials{
+		CredentialsID:   credentialsID,
+		AccessKeyID:     accessKeyID,
+		SecretAccessKey: secretAccessKey,
+		SessionToken:    sessionToken,
+	}
+
+	testTask := testdata.LoadTask("sleep5")
+	// Configure the task and container to use execution role
+	testTask.SetExecutionRoleCredentialsID(credentialsID)
+	testTask.Containers[0].RegistryAuthentication = &api.RegistryAuthenticationData{
+		Type: "ecr",
+		ECRAuthData: &api.ECRAuthData{
+			UseExecutionRole: true,
+		},
+	}
+	container := testTask.Containers[0]
+
+	credentialsManager.EXPECT().GetTaskCredentials(credentialsID).Return(credentials.TaskIAMRoleCredentials{
+		ARN:                "",
+		IAMRoleCredentials: executionRoleCredentials,
+	}, true)
+
+	client.EXPECT().PullImage(gomock.Any(), gomock.Any()).Do(
+		func(image string, auth *api.RegistryAuthenticationData) {
+			assert.Equal(t, container.Image, image)
+			assert.Equal(t, auth.ECRAuthData.GetPullCredentials(), executionRoleCredentials)
+		}).Return(DockerContainerMetadata{})
+	imageManager.EXPECT().RecordContainerReference(container).Return(nil)
+	imageManager.EXPECT().GetImageStateFromImageName(container.Image)
+
+	taskEngine.(*DockerTaskEngine).pullContainer(testTask, container)
+}
+
+// TestTasktionRoleOnRestart tests the agent will process the task recorded in
```

Comment:
TestNewTaskTransitionOnRestart tests that the agent will process the task recorded in the state file on restart.